daemon/crypto/secure/secure.go

```diff
@@ -0,0 +1,138 @@
+// Package secure is a wrapper package around memguard for managing data which
+// must not be swapped to disk or easily read via a memory scanner.
+package secure
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/awnumar/memguard"
+)
+
+// ErrSecretNotFound occurs when a secret could not be found
+var ErrSecretNotFound = errors.New("Could not find secret to remove it")
+
+var lock *sync.Mutex
+var current *Guard
+
+func init() {
+	lock = &sync.Mutex{}
+	memguard.DisableUnixCoreDumps()
+}
+
+// Guard is a controller of guarded memory which contains many different
+// secrets. Only one guard can exist at a time.
+type Guard struct {
+	secrets []*Secret
+	lock    *sync.Mutex
+}
+
+// Secret represent a specific guraded piece of memory
+type Secret struct {
+	guard  *Guard
+	buffer *memguard.LockedBuffer
+}
+
+// NewGuard returns a Guard instance or errors if a guard already exists.
+func NewGuard() *Guard {
+	lock.Lock()
+	defer lock.Unlock()
+
+	if current != nil {
+		return current
+	}
+
+	current = &Guard{
+		secrets: []*Secret{},
+		lock:    &sync.Mutex{},
+	}
+
+	return current
+}
+
+func (g *Guard) remove(s *Secret) error {
+	if s == nil {
+		panic("received a nil secret; should not be possible")
+	}
+
+	g.lock.Lock()
+	defer g.lock.Unlock()
+
+	placement := -1
+	for i, v := range g.secrets {
+		if *v == *s {
```

Comment:
Did you mean to dereference `v` and `s` here? Instead of comparing the addresses of the structs, the contents are being compared here (which is not a huge deal *in this case*, but it seems unnecessary).

Reply:
Great catch; what I really want here is to find if the secret points to the same underlying LockedBuffer instance. I'll open a pr with a fix!

Reply:
@ianlivingstone Ah, so you want to do `v.buffer == s.buffer`. But assuming that only a single secret holds a specific LockedBuffer instance, you should get away with `v == s`.

Reply:
Yep! Just opened a PR with the change: https://github.com/manifoldco/torus-cli/pull/326

Thanks again for the catch :)